src/files-and-videos/videos-page/data/thunks.js

```diff
@@ -51,6 +50,19 @@ export function fetchVideos(courseId) {
       }));
       dispatch(setPageSettings({ ...data }));
       dispatch(updateLoadingStatus({ courseId, status: RequestStatus.SUCCESSFUL }));
+      parsedVideos.forEach(async (video) => {
```

Comment:
There is a problem with using `forEach` and `await` together, since it looks like it should do something other than what it does. You are probably aware of this. The await only applies inside each loop, but the next iteration starts before the first one finishes. So basically this looks like a loop that fires requests sequentially, but in truth it fires them in parallel. Which is probably exactly what you want.

I would probably recommend to get rid of the `async` in the forEach line so it doesn't create any confusion and instead wrap everything inside into an inner async function that you call without an await.
I mean something like:

```
async function fetchAndUpdateVideoUsagePaths(<...>) {
  const { usageLocations } = await getVideoUsagePaths({ videoId: video.id, courseId });
  // rest of logic ...
}

parsedVideos.forEach((video) => {
  fetchAndUpdateVideoUsagePaths(<...>)
})
```